pcap-linux.c

```diff
@@ -3765,65 +3216,13 @@ prepare_tpacket_socket(pcap_t *handle)
 		/*
 		 * Success.
 		 */
-		return 1;
-	}
-	if (ret == -1) {
-		/*
-		 * We failed for some reason other than "the
-		 * kernel doesn't support TPACKET_V2".
-		 */
-		return -1;
+		return 0;
```

Comment:
This, however, should fail even if `init_tpacket()` returns 1 (that "should not happen").